internal/pkg/build/buildkit/daemon/daemon.go

```diff
@@ -198,6 +206,72 @@ func Run(ctx context.Context, socketChan chan<- string) error {
 	return err
 }
 
+func setDefaultConfig(cfg *config.Config) {
+	orig := *cfg
+
+	// If we need to, enter a new cgroup now, to workaround an issue with crun container cgroup creation (#1538).
+	if err := oci.CrunNestCgroup(); err != nil {
+		sylog.Fatalf("while applying crun cgroup workaround: %v", err)
+	}
+
+	rlUID, err := rootless.Getuid()
+	if err != nil {
+		sylog.Fatalf("While trying to determine uid: %v", err)
+	}
+	if rlUID != 0 {
+		cfg.Workers.OCI.Rootless = true
+		cfg.Workers.OCI.NoProcessSandbox = true
+	}
+
+	if cfg.GRPC.UID == nil {
+		uid := os.Getuid()
+		cfg.GRPC.UID = &uid
+	}
+
+	if cfg.GRPC.GID == nil {
+		gid := os.Getgid()
+		cfg.GRPC.GID = &gid
+	}
+
+	enabled := true
+	cfg.Workers.OCI.Enabled = &enabled
+
+	if cfg.Root == "" {
+		cfg.Root = appdefaults.Root + "-singularity-ephemeral"
+	}
+
+	cfg.Workers.OCI.Snapshotter = "overlayfs"
+
+	if cfg.Workers.OCI.Platforms == nil {
+		cfg.Workers.OCI.Platforms = formatPlatforms(archutil.SupportedPlatforms(false))
+	}
+	if cfg.Workers.Containerd.Platforms == nil {
+		cfg.Workers.Containerd.Platforms = formatPlatforms(archutil.SupportedPlatforms(false))
+	}
+
+	sylog.Debugf("cfg.Workers.OCI.Platforms: %#v", cfg.Workers.OCI.Platforms)
+
+	cfg.Workers.OCI.NetworkConfig = setDefaultNetworkConfig(cfg.Workers.OCI.NetworkConfig)
+	cfg.Workers.Containerd.NetworkConfig = setDefaultNetworkConfig(cfg.Workers.Containerd.NetworkConfig)
```

Comment:
Do we need to initialize this, given we only use OCI workers?

Reply:
Removing this doesn't drop any dependencies, it turns out, but still probably a good idea to remove it regardless 👍